tools/codegen/api/python.py

```diff
@@ -304,15 +395,54 @@ def output_idx(self) -> int:
     # tools/gen_pyi.py.  If you change any logic here, please
     # check that file too.
```

Comment:
the old comment can be changed as the logic has been moved to here :) (unless there are hard coded ones left in gen_pyi still to be checked?)